admin_ui/src/components/MessagePopup.vue

```diff
@@ -1,7 +1,22 @@
 <template>
     <div
-        id="message_popup"
-        v-if="visible"
+        id="message_error"
+        v-if="visible && message.includes('error')"
+    >
+        <p class="message">{{ message }}</p>
+        <p class="close">
+            <a
+                class="subtle"
+                href="#"
+                v-on:click.prevent="visible = false"
+            >
+                <font-awesome-icon icon="times" />
+            </a>
+        </p>
+    </div>
+    <div
+        id="message_success"
+        v-else-if="visible"
```

Comment:
Rather than duplicating the code, it might be cleaner to have a dynamic class or style applied which just sets the background red.

Something like:

```
:class="{error: apiResponseMessage.type == 'error'}"
```

Reply:
Like this

```Vue
<template>
    <div
        id="message_popup"
        v-bind:style="{'background-color':'red'}"
        v-if="visible && apiResponseMessage.type == 'error'"
    >
        <p class="message">{{ message }}</p>
        <p class="close">
            <a
                class="subtle"
                href="#"
                v-on:click.prevent="visible = false"
            >
                <font-awesome-icon icon="times" />
            </a>
        </p>
    </div>
    <div
        id="message_popup"
        v-else-if="visible"
    >
        <p class="message">{{ message }}</p>
        <p class="close">
            <a
                class="subtle"
                href="#"
                v-on:click.prevent="visible = false"
            >
                <font-awesome-icon icon="times" />
            </a>
        </p>
    </div>
</template>


<script lang="ts">
import Vue from "vue"
import * as i from "../interfaces"

export default Vue.extend({
    data() {
        return {
            visible: false,
            timeLastAppeared: 0,
        }
    },
    computed: {
        message(): string {
            return this.apiResponseMessage
                ? this.apiResponseMessage.contents
                : "-"
        },
        apiResponseMessage(): i.APIResponseMessage {
            return this.$store.state.apiResponseMessage
        },
    },
    methods: {
        getTime(): number {
            return new Date().getTime()
        },
    },
    watch: {
        visible() {
            this.timeLastAppeared = this.getTime()
        },
        apiResponseMessage() {
            this.visible = true
            const app = this

            setTimeout(() => {
                const now = this.getTime()
                // Only hide if it if it wasn't subsequently opened again.
                if (now - app.timeLastAppeared >= 3000) {
                    app.visible = false
                }
            }, 3000)
        },
    },
})
</script>


<style lang="less">
@import "../vars.less";

div#message_popup {
    display: flex;
    position: fixed;
    bottom: 0;
    width: 100%;
    padding: 0.5rem 1rem;
    box-sizing: border-box;
    background-color: green;

    p,
    a {
        color: white !important;
    }

    p.message {
        flex-grow: 1;
        text-align: center;
    }

    p.close {
        flex-grow: 0;
    }
}
</style>
```

Reply:
Or even better without code duplication

```Vue
<template>
    <div
        id="message_popup"
        v-bind:style="[apiResponseMessage.type == 'error' ? {'background-color':'red'} : {'background-color':'green'}]"
        v-if="visible"
    >
        <p class="message">{{ message }}</p>
        <p class="close">
            <a
                class="subtle"
                href="#"
                v-on:click.prevent="visible = false"
            >
                <font-awesome-icon icon="times" />
            </a>
        </p>
    </div>
</template>
```

Reply:
Yes, that's nice. I slightly prefer using a `:class`, just because a CSS variable can be used for the colour.

In `variables.less` there's a `@red` value - so this way, we can use that, and it keeps the colour consistent with the rest of the app.